### PR TITLE
[office-js-preview] (Outlook) Add multi-select method and event

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [Conceptual doc](https://github.com/OfficeDev/office-js-docs-pr/pull/3675)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -2413,6 +2413,15 @@ declare namespace Office {
          * Triggers when a Resource selection happens in Project.
          */
         ResourceSelectionChanged,
+        /**
+         * Occurs in Outlook when one or more messages are selected or deselected.
+         * **Important**: Only available with task pane implementation.
```
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts:2418]: This Important note seems ambiguous to me. Does it mean that the event is only triggered/handled in a JavaScript in a task pane; that is, _not_ in a function command? If not, what exactly is meant?

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -14493,6 +14502,54 @@ declare namespace Office {
...
+         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter,
+         *        `asyncResult`, which is an `Office.AsyncResult` object. The properties of the selected messages are returned as an array of JSON objects in the
+         *        `asyncResult.value` property. These properties include the item ID, subject, item type (Message is the only supported type at this time), and item mode
+         *        (Read or Compose). The objects in the array follow the order in which messages were selected.
+         * 
+         * @beta
+         */
+        getSelectedItemsAsync(options: Office.AsyncContextOptions, callback: (asyncResult: Office.AsyncResult<object[]>) => void): void;
+        /**
+         * Gets currently selected messages on which an add-in can activate and perform operations. An add-in can activate on a maximum of 100 messages at a time.
```
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts:14530]: It's not clear to me what "activate" means here. Does this feature require the use of activation rules, such as described in https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/activation-rules ?

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

File: types/office-js-preview/index.d.ts (hunk 2 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts]: In this context, "activate" refers to the add-in "lighting up"/becoming active in the ribbon when multiple messages are selected (as opposed to it appearing grayed out). Multi-select doesn't require the use of activation rules.

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:2418]: Thanks for the feedback, @Rick-Kirkham. You're correct; the event can only be handled in a task pane and not in a function command. How about something like, "This event can only be handled in a task pane. It isn't supported in function commands."?

File: types/office-js-preview/index.d.ts (hunk 2 above)
Comment (Rick-Kirkham) [reply types/office-js-preview/index.d.ts]: got it.

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (Rick-Kirkham) [reply types/office-js-preview/index.d.ts:2418]: Sounds great.

**Review by Rick-Kirkham — APPROVED** [review]

**Pull request merged**